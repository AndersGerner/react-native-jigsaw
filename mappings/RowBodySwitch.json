[{"name":"Row Single Line Body Switch","tag":"RowBodySwitch","description":"A row with left aligned body text and a right aligned switch","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Row_SingleLine_BodyToggle.png","category":"deprecated","supports_list_render":true,"props":{"title":{"label":"Title","description":"Text to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"color":{"label":"Color","description":"Custom color for switch","editable":true,"required":false,"type":"color"},"fieldName":{"label":"Field name","description":"The name of the field within the screen that will store this component's value","type":"fieldName","value":"switchValue","valuePropName":"value","handlerPropName":"onValueChange","editable":true,"required":false}},"layout":{}},{"name":"Row Double Line Body Switch","tag":"RowBodySwitch","description":"A row with left aligned body text and subtitle text and a right aligned switch","category":"deprecated","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Row_DoubleLine_Body_Toggle.png","supports_list_render":true,"props":{"title":{"label":"Title","description":"Text to display","type":"string","value":"Beautiful West Coast Villa","editable":true},"subtitle":{"label":"Subtitle","description":"Subtitle text to display","type":"string","value":"San Diego","editable":true},"color":{"label":"Color","description":"Custom color for switch","editable":true,"required":false,"type":"color"},"fieldName":{"label":"Field name","description":"The name of the field within the screen that will store this component's value","type":"fieldName","value":"switchValue","valuePropName":"value","handlerPropName":"onValueChange","editable":true,"required":false}},"layout":{}}]