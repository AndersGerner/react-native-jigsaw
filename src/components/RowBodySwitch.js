/* @flow */

import * as React from "react";
import { withTheme } from "../core/theming";
import { COMPONENT_TYPES, FORM_TYPES } from "../core/component-types";
import Row from "./Row";
import Switch from "./Switch";
import type { Theme } from "../types";

type Props = {
  title: string,
  subtitle?: string,
  style?: any,
  theme: Theme,
  value?: boolean,
  color?: string,
  onValueChange?: Function
};

class RowBodySwitch extends React.Component<Props> {
  render() {
    const {
      title,
      subtitle,
      style,
      value,
      onValueChange,
      color,
      theme: { colors, typography, spacing }
    } = this.props;

    return (
      <Row
        titleTypeStyle={typography.body1}
        titleColor={colors.medium}
        subtitleTypeStyle={typography.subtitle2}
        subtitleColor={colors.light}
        title={title}
        subtitle={subtitle}
        right={() => (
          <Switch value={value} color={color} onValueChange={onValueChange} />
        )}
        style={style}
      />
    );
  }
}

export default withTheme(RowBodySwitch);

export const SEED_DATA = [
  {
    name: "Row Single Line Body Switch",
    tag: "RowBodySwitch",
    description: "A row with left aligned body text and a right aligned switch",
    preview_image_url:
      "https://res.cloudinary.com/altos/image/upload/v1541096679/draftbit/library/jigsaw-1.0/reps/Row_SingleLine_BodyToggle.png",
    category: COMPONENT_TYPES.row,
    supports_list_render: true,
    props: {
      title: {
        label: "Title",
        description: "Text to display",
        type: FORM_TYPES.string,
        value: "Beautiful West Coast Villa",
        editable: true
      },
      value: {
        label: "Status",
        description: "boolean status if switch is toggled true or false",
        editable: true,
        value: true,
        type: FORM_TYPES.boolean
      },
      onValueChange: {
        label: "onValueChange handler",
        description: "Function to change value of the switch",
        editable: true,
        required: true,
        value: "{this.onValueChange}",
        type: FORM_TYPES.function
      },
      color: {
        label: "Color",
        description: "Custom color for switch",
        editable: true,
        required: false,
        type: FORM_TYPES.color
      }
    },
    layout: {
      width: 375,
      height: 59
    }
  },
  {
    name: "Row Double Line Body Switch",
    tag: "RowBodySwitch",
    description:
      "A row with left aligned body text and subtitle text and a right aligned switch",
    category: COMPONENT_TYPES.row,
    preview_image_url:
      "https://res.cloudinary.com/altos/image/upload/v1541096697/draftbit/library/jigsaw-1.0/reps/Row_DoubleLine_Body_Toggle.png",
    supports_list_render: true,
    props: {
      title: {
        label: "Title",
        description: "Text to display",
        type: FORM_TYPES.string,
        value: "Beautiful West Coast Villa",
        editable: true
      },
      subtitle: {
        label: "Subtitle",
        description: "Subtitle text to display",
        type: FORM_TYPES.string,
        value: "San Diego",
        editable: true
      },
      value: {
        label: "Value",
        description: "Boolean status if switch is toggled true or false",
        editable: true,
        value: true,
        type: FORM_TYPES.boolean
      },
      onValueChange: {
        label: "onValueChange handler",
        description: "Function to change value of the switch",
        editable: true,
        required: true,
        value: "{this.onValueChange}",
        type: FORM_TYPES.function
      },
      color: {
        label: "Color",
        description: "Custom color for switch",
        editable: true,
        required: false,
        type: FORM_TYPES.color
      }
    },
    layout: {
      width: 375,
      height: 81
    }
  }
];
