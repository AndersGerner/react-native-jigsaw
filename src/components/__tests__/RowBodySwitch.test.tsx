import * as React from "react";
import renderer from "react-test-renderer";
import RowBodySwitch from "../RowBodySwitch";

it("renders row body switch", () => {
  const tree = renderer.create(<RowBodySwitch value={true} />).toJSON();

  expect(tree).toMatchSnapshot();
});
