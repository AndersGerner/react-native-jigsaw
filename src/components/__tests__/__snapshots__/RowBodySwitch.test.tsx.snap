// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders row body switch 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Object {
        "padding": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 26,
            },
            Object {
              "color": "rgba(70, 78, 88, 1)",
            },
          ]
        }
      />
    </View>
  </View>
  <RCTSwitch
    accessibilityRole="switch"
    activeTrackColor="rgba(90, 69, 255, 1)"
    onChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTintColor="rgba(90, 69, 255, 1)"
    style={
      Array [
        Object {
          "height": 31,
          "width": 51,
        },
        Array [
          Object {
            "opacity": 1,
          },
          undefined,
        ],
      ]
    }
    tintColor=""
    value={true}
  />
</View>
`;
